### ISSUE TITLE
3.0 Release prep

### ISSUE THREAD

**Issue by phkahler** [solvespace/solvespace#772]
Pulling all ideas from comments up here.
Like the checkbox approach - Github will just update this comment when checked:

## Prior to any public builds:
  * [x] Decide whether we want to keep the git hash in the snap version for tagged commits (YES)
  * [x] Update version strings? Any in exported files - Step supports this.
  * [x] Add (an issue for) documentation for the new features
  * [x] Update the reference manual and website
  * [x] Decide what WIndows gets. there are 2 versions (with OpenMP and without)
  * [ ] Possibly make a windows installer (using NSIS?) - only if someone wants to make that happen.
## After beta or RC:
  * [x] Call for updating translations - even en_US reports missing translations.
  * [ ] Test the binaries ;)
  * [x] Make sure Windows and macOS builds have no shared libraries?
  * [x] Test the macOS build on a 'non-infected' computer, preferably with an older OS version?
  * [x] Make sure the cli is also working (we could test this in CI as well)
  * [x] Make sure macOS notarising is working (I saw a warning about a secret not being escaped, I will see what happens with the current build and if it fails I will re-upload the secrets)
  * [ ] SolveSpace as a library.  Linux builds libsolvespace-headless.a  PR #493 suggests it still works.
## After release:
  * [ ] Promote the stable, tagged revision in the Snap Store to channel stable
  * [ ] Propose solvespace for "Featured Snaps Friday" (https://forum.snapcraft.io/t/call-for-suggestions-featured-snaps-friday-23rd-october-2020/20637)
  * [x] See if we can get a mention on Phoronix.com (not sure if he takes submissions).

**Comment by phkahler** [comment]
Updated version strings anywhere?

**Comment by ruevs** [comment]
Once the release is ready - update the web site - conservatively - but I do have a few things in mind. Once the GitHub project is updated does anyone know how to put it up on solvespace.com? Maybe @jwesthues still wants to do this himself?

**Comment by jwesthues** [comment]
There's a git repo for the website in this same organization; feel free to make the changes. Let me know when you're ready to deploy, and I can pull to live.

**Comment by vespakoen** [comment]
Some things that come to mind:

- Test the binaries ;)
- Make sure Windows and macOS builds have no shared libraries?
- Test the macOS build on a 'non-infected' computer, preferably with an older OS version?
- Make sure the cli is also working (we could test this in CI as well)
- Make sure macOS notarising is working (I saw a warning about a secret not being escaped, I will see what happens with the current build and if it fails I will re-upload the secrets)
- Add (an issue for) documentation for the new features
- SolveSpace as a library, is this still possible / being built? (http://solvespace.com/library.pl)
- Possibly make a windows installer (using NSIS?) that also includes the OpenMP dll?

**Comment by ppd** [comment]
Before & after release:
- [ ] Decide whether we want to keep the git hash in the snap version for tagged commits
- [ ] Promote the stable, tagged revision in the Snap Store to channel `stable`
- [ ] Propose solvespace for "Featured Snaps Friday" (https://forum.snapcraft.io/t/call-for-suggestions-featured-snaps-friday-23rd-october-2020/20637)

**Comment by phkahler** [comment]
@ppd Decide whether we want to keep the git hash in the snap version for tagged commits

Is there a reason not to keep that?

**Comment by ppd** [comment]
@phkahler just optics, I guess. Stable could be just "3.0". On the other hand, the hash is very useful for quickly mapping a revision to a commit, so lets keep it.

**Comment by phkahler** [comment]
@vespakoen  Test the macOS build on a 'non-infected' computer, preferably with an older OS version?

It doesn't work on my Air with 10.12.6 Sierra:
Dyld Error Messaage:
  Library not loaded: /usr/local/opt/llvm@9/lib/libomp.dylib
  Referenced from: /Volumes/VOLUME/SolveSpace/app/Contents/MacOS/SolveSpace
  Reason: image not found

There is no llvm@9 folder on my machine.

**Comment by vespakoen** [comment]
Yep I also noticed, I am working on a fix...

**Comment by vespakoen** [comment]
@phkahler can you try with the latest edge? would be interesting to see if it works on Sierra

**Comment by phkahler** [comment]
> @phkahler can you try with the latest edge? would be interesting to see if it works on Sierra

It seems to work. I open the DMG and it does whatever it does and a folder opens with just SolveSpace. Running that gives me one pop-up about opening random stuff from the internet. I say open anyway and it seems to just work. I'll give a better going over later when we're a little more finalized.

**Comment by vespakoen** [comment]
Yeah that sounds correct, we can improve the installation method on macOS by adding a symbolic link to the Applications folder and include a background, but the process for this is... hairy and complicated to say the least.

The warning for stuff downloaded from the internet is also expected, the only way around this is to release through the App Store, not sure if this is worth it though...

**Comment by phkahler** [comment]
Is the Windows build 32 or 64bit?

**Comment by jwesthues** [comment]
> Is the Windows build 32 or 64bit?

I guess if we offer a 64-bit version, then it's still customary to also offer a 32-bit version? So the lazy option would be 32 only. I'm not sure if anyone's ever benchmarked, but I doubt there'd be a big speed advantage to the 64.

**Comment by ppd** [comment]
@phkahler `-DCMAKE_GENERATOR_PLATFORM="Win32"` ⇒ 32-bit

**Comment by ruevs** [comment]
32 bit only in my opinion. SolveSpace will be unusably slow long before it consumes 2GB of RAM, which is the limit for 32bit Win32 processes. At this point there is absolutely no advantage to a 64bit build (but it does work).

**Comment by phkahler** [comment]
I just wanted to know which version of the vc_redist stuff to mention for the OMP build ;-) x86 it is.

**Comment by phkahler** [comment]
Are we good to make an RC1? Once we know what to link to I (someone) can add a table entry on the download page and have JW push the changes there.

**Comment by ruevs** [comment]
To me master  looks the best so far (which admittedly it is most of the time), all the milestone issues are done. I think an RC1 linked from the official page is in order. Many more people need to run it to find more subtle bugs. So I am all for it.

So what is the procedure? Make a "3.0_RC" branch, change the version there and tag it? Then wait for Travis to build it?

**Comment by phkahler** [comment]
I will defer to @ppd but I don't think a branch prior to 3.0 is really warranted. A branch at 3.0 might be nice in case we want to back port important fixes once other more invasive changes start happening. BTW @ppd if you're in agreement feel free to pull the trigger.

**Comment by jwesthues** [comment]
Yeah--I'd probably make a branch at 3.0 in case we need to backport, and tag the release candidate(s). We should test the RC among ourselves, then post it on the download page (clearly labeled as experimental/beta/whatever), and then if all seems well after a couple months relabel it as the latest official and recommended release.

**Comment by ppd** [comment]
My personal preference: 
1. tag master HEAD `v3.0.rc1` right now
2. polish the release on master
3. tag master HEAD `v3.0`
4. branch there to `3.x`

So a yes to the workflow of @phkahler & @jwesthues.

**Comment by ruevs** [comment]
Since we do not have anything "urgent" with significant risky changes to do on "master" @ppd 's suggestion to "stabilize" (do the RCs) on master should be fine. It will save some potential merging back-and forth. But we will have to avoid doing risky/drastic/major things on master until we release 3.0 and branch 3.x

**Comment by phkahler** [comment]
@ppd  time for an RC. Can you kick that off and see that it arrives on the download page OK?  Then we'll have JW update the web site.

**Comment by ppd** [comment]
Alright. It needs a little Travis fix, but then we should be good to go.

**Comment by phkahler** [comment]
Yeah it put source code but not executables ;-)

@jwesthues  Can you make the recent website changes live? Oh, and I guess we need to add another table entry for the 3.0-RC1 release - I'm not able to do that for another 10 hours so that would be good too.

**Comment by ppd** [comment]
Okay. Seems like that worked except for that weird snap glitch.  I think it has to do with the fact that it's a tagged commit. Most likely a braindead error in `snapcraft.yaml`. I'll investigate.

**Comment by jwesthues** [comment]
I've added the table entry, and pulled to live. Thanks, and let me know if anyone sees any issues.

**Comment by ruevs** [comment]
In the "Platforms" column - (unfortunately) it does not work on XP any more. So it should be  "Windows Vista–10, Linux, Mac"

It does still work on XP if one compiles the OpenGL 1 front end version and goes through some extra hoops with the SDK and compiler. Probably I'll make an unofficial XP release on my fork when 3.0 is final.

**Comment by 0x0203** [comment]
Not sure if you want this here or in a separate issue, but building (for Linux) from the packaged solvespace-3.0.rc1.tar.gz on the downloads page doesn't work since it doesn't include the required submodules. I don't know if it's supposed to, but cmake won't complete without q3d, libdxfrw, mimalloc, and flatbuffers. And since this isn't a git clone, running the 'git submodule update --init ...' command won't work.

Incidentally, the build instructions for Linux only mention the git submodule update step for libdxfrw. Including mimalloc, flatbuffers and q3d if they really are required might be useful.

**Comment by jwesthues** [comment]
@ruevs Thanks, good catch, merged and should be live.

**Comment by Symbian9** [comment]
### Before 3.0 release:

- [ ] Revert https://github.com/solvespace/solvespace/commit/b2418a0324fa4e4ed498a46aa0946b67a267ebf0
- [ ] Request removing of https://cla-assistant.io/solvespace/solvespace

**Comment by ruevs** [comment]
> ### Before 3.0 release:
>     * [ ]  Revert [b2418a0](https://github.com/solvespace/solvespace/commit/b2418a0324fa4e4ed498a46aa0946b67a267ebf0)
>     * [ ]  Request removing of https://cla-assistant.io/solvespace/solvespace

@Symbian9 3.0 does not depend on these at all.

**Comment by phkahler** [comment]
Should we have an RC2?

I'd really like to get 3.0 Final, but if we want an RC2 I'd probably merge the rebased Eigen stuff, the Apple M1 if it gets there, look into the SpaceMouse issue, and one last minute feature I'd like, and I think @ruevs tagged something for 3.0.

**Comment by Symbian9** [comment]
> Should we have an RC2?

Yes! Especially to give more time for translator to update/add new locales — `3.0rc2` should be *"Call for Translators!"*.

**Comment by Symbian9** [comment]
> **Before 3.0 release:**
> 
> * [ ]  Revert [b2418a0](https://github.com/solvespace/solvespace/commit/b2418a0324fa4e4ed498a46aa0946b67a267ebf0)
> * [ ]  Request removing of https://cla-assistant.io/solvespace/solvespace

> 3.0 does not depend on these at all.

Removing CLA signing requirement for contributing *before* `3.0` is really important!

@ruevs, please, read comment by @jwesthues:

 - https://github.com/solvespace/solvespace/issues/816#issuecomment-735278731

**Comment by rpavlik** [comment]
So the Debian freeze for Bullseye is rapidly approaching (first freeze is already past) - a Debian Science maintainer is updating the package, but since rc1 was the last tag, that's what's getting packaged. It would be great to get an RC2 soon if nothing else but for getting into Debian, since it's upstream of so much.

Looks like there's currently a build failure on armel due to libdxfrw? https://buildd.debian.org/status/fetch.php?pkg=solvespace&arch=armel&ver=3.0.rc1%2Brepack1-4&stamp=1610723096&raw=0 The overall status page is here: https://tracker.debian.org/pkg/solvespace

**Comment by phkahler** [comment]
The only thing I need prior to RC2 is to update the .pot file for translations. I can maybe do that tonight or this weekend, or anyone else can do it and make a PR which I can merge from phone. I want that step done so translators only need to update their particular locale after the RC.

In an ideal world we'd figure out the one remaining GTK oddness with a 3.0 label, and the SpaceMouse on Windows. But the GTK one is actually (I think) an old problem so we can leave it. SpaceMouse would be a shame to lose even though it seems uncommon with FLOSS tool users.

After RC2 we're looking at just translations and maybe some icon updates.

**Comment by Symbian9** [comment]
> The only thing I need prior to RC2 is to update the .pot file for translations. I can maybe do that tonight or this weekend, or anyone else can do it and make a PR which I can merge from phone.

I'm ready for updating Ukrainian locale!

**Comment by rpavlik** [comment]
I tried to do the translations update, but I'm new at it so if I goofed it up  just let me know or close it.

**Comment by ruevs** [comment]
> So the Debian freeze for Bullseye is rapidly approaching (first freeze is already past) - a Debian Science maintainer is updating the package, but since rc1 was the last tag, that's what's getting 

Definitely worth making a rc2 for this!

It's impressive (but not surprising) how much "infrastructure" is involved in Debian packages!
https://salsa.debian.org/science-team/solvespace/-/commits/master/

From the above it seems there may be a mimalloc cmake change that we may want to take? To use the CXX instead of the C compiler?


> Looks like there's currently a build failure on armel due to libdxfrw? https://buildd.debian.org/status/fetch.php?pkg=solvespace&arch=armel&ver=3.0.rc1%2Brepack1-4&stamp=1610723096&raw=0 The overall status page is here: https://tracker.debian.org/pkg/solvespace

I can not understand what failed... I'll look again later on a computer...

**Comment by ruevs** [comment]
Making rc2 will also reduce the Debian package by one patch
https://salsa.debian.org/science-team/solvespace/-/blob/master/debian/patches/20_use_system_flatbuffers.patch
Since we removed flatbuffers.

**Comment by rpavlik** [comment]
Yes, I saw that patch when I reviewed the state of the package, and thought the same thing. That will simplify things indeed.

Once the package passes all the Debian arches, we can upstream the patches that are appropriate, if any. I think the CXX mimalloc  change was reverted but not 100% sure: I had pinged that dev to ask if he'd have time to look at this package, and if not, I'd do it. Didn't get a reply, but saw he started in on it, so I left him to it rather than trying to race.

**Comment by phkahler** [comment]
I believe we are ready for RC2.  @ppd or @rpavlik ?

**Comment by rpavlik** [comment]
Seems ok to me. I filled an issue with the mailing list email I got today about it, seems there is a build issue with mimalloc on arm

**Comment by ppd** [comment]
> I believe we are ready for RC2. @ppd

Yes, go for it.

**Comment by ruevs** [comment]
@phkahler only consider the suggestion here:
https://github.com/solvespace/solvespace/issues/890#issuecomment-761824614
Do you guys think that "Fade Previous Solid Groups" is better?

**Comment by Symbian9** [comment]
> Do you guys think that "Fade Previous Solid Groups" is better?

Yes!

But also built-in style `s00d-#def-dim-solid` should be renamed to `s00d-#def-fade-solid`

**Comment by rurban** [comment]
No opinion on def or not. Both sound fine to me. The shorter might be a bit more appealing.

**Comment by phkahler** [comment]
@ppd I tried the "Draft a new Release" button, filled in a name and a couple words, checked the checkbox (this is a pre-release). And nothing happened. What's the correct procedure?

**Comment by ruevs** [comment]
Give me a few more minutes... I'm preparing the non anti-aliased icons...

**Comment by ruevs** [comment]
@phkahler https://github.com/solvespace/solvespace/pull/897

**Comment by ppd** [comment]
@phkahler That's the correct procedure. Did you enter a tag name?

**Comment by phkahler** [comment]
> @phkahler That's the correct procedure. Did you enter a tag name?

I did. You want to give it a go?

**Comment by phkahler** [comment]
@jwesthues   I updated solvespace-web to point at the RC2. Can you make that live?

Now it's just translations and icons ;-)

**Comment by jwesthues** [comment]
Excellent, just pulled to live.

**Comment by phkahler** [comment]
Oops. I didnt update the text, just the table and link.

**Comment by jwesthues** [comment]
No worries, let me know if you need anything else.

**Comment by rpavlik** [comment]
I have made debian package changes to include rc2 and the patch for mimalloc, as well as for being able to use system three.js (needed some changes to go from  r76 to r111, will upstream those eventually). Have reached out to the person who did the maintainance of the package last week requesting review/sponsorship. It seems to work when I use it...

**Comment by Symbian9** [comment]
>  I have made debian package changes... Have reached out **to the person who did the maintainance of the package** last week requesting review/sponsorship

Are you mean @AlexanderP?

- https://notesalexp.org/packages/en/buster/amd64/solvespace/

**Comment by rpavlik** [comment]
No, Anton Gladky, debian project member. alexp doesn't appear to upstream their packages?

**Comment by Symbian9** [comment]
> alexp doesn't appear to upstream their packages?

He regularly updated builds in his PPA & use his builds.

**Comment by rpavlik** [comment]
I looked at his packages: There was some stuff I was able to use in the Debian package, but in general they look mostly separate (the changelog is fully distinct, anyway) and not quite meeting debian policy, etc.

**Comment by rpavlik** [comment]
Woohoo, so we got rc2 (plus some patches) into Debian Testing in time! https://tracker.debian.org/news/1224251/solvespace-30rc2repack1-1-migrated-to-testing/

**Comment by phkahler** [comment]
@rpavlik   So I tried to run:
```
git -C extlib/mimalloc pull https://github.com/microsoft/mimalloc v2.0.0
```
And got this:   fatal: refusing to merge unrelated histories
Tips on obscure git stuff welcome ;-)

**Comment by vespakoen** [comment]
```sh
cd extlib/mimalloc
git checkout master
```

Should do the trick

**Comment by phkahler** [comment]
@vespakoen   no, that puts me back on my solvespace master branch. I want to bump the mimalloc version we use to their v2.0.0

**Comment by rpavlik** [comment]
oh, did we fork their stuff? I'll glance at it shortly.

**Comment by rpavlik** [comment]
ahhh - there is m1 stuff post-2.0.. can't get it all I guess 
![image](https://user-images.githubusercontent.com/61129/107129215-6fc63100-6889-11eb-84fa-648491557102.png)

**Comment by phkahler** [comment]
I think we used something from the dev branch before.

**Comment by rpavlik** [comment]
OK, I got it to update. Not sure if we want the dev branch patches too or not, but I have a PR now for 2.0.0. @vespakoen 's stuff worked for me, I'm assuming there was an edit that crossed in the mail.

**Comment by phkahler** [comment]
Thank you!

**Comment by phkahler** [comment]
I'm ready to call 3.0 done. Any objections? I'll give it a day or three.

**Comment by Symbian9** [comment]
> Any objections? I'll give it a day or three.

We need just a little time to update locales for `3.0`!

Maybe update `messages.pot` template now?

- https://github.com/solvespace/solvespace/blob/master/res/messages.pot

**Comment by phkahler** [comment]
I'm not waiting for translations any more. No new features. No bug fixes.

If it builds on all platforms, as I think it will, passes tests, and there are no regressions from recent pull requests then its done.

**Comment by ruevs** [comment]
I'm all for it.

Of course I'm sorely tempted by the IdList optimizations (https://github.com/solvespace/solvespace/pull/992, [FixIdList_TheNextAttempt](https://github.com/ruevs/solvespace/commits/FixIdList_TheNextAttempt)), but that's extremely unwise. It's rather unlikely that we won't break anything.

So let's release a very **stable**, the "best so far" 3.0.

Then we can quickly make a **fast** 3.1 rc 1 and use the "excitement" generated by 3.0 to regression test it (by hopefully many users).

**Comment by ruevs** [comment]
@phkahler do you think I should change the extrude icon as suggested here   https://github.com/solvespace/solvespace/issues/857#issuecomment-804415270 ?

**Comment by pjanx** [comment]
Releases are mostly for distributions, and the jump in quality since 2.3 is huge. It should get out of the door ASAP, so long as there are no clear regressions against the 2.3. I'm fairly certain #992 is safe, but it doesn't actually matter. The lesser version we've managed to get in is more than sufficient.

**Comment by pjanx** [comment]
And, since I'm already commenting here, CMake has support for making NSIS packages, but I haven't had good experience with it. Namely, there was an issue with having multiple installations of the same package--they conflicted. And it [just wasn't very nice](https://git.janouch.name/p/logdiag/src/branch/master/CMakeLists.txt#L415). Maybe it's got better since then.

**Comment by ruevs** [comment]
An installer (on Windows of course) has been discussed - it's a "no" from most contributors.

A single exe with no dependencies does not need an installer. The MSVC runtime distributable dependency of the OpenMP build is already installed on 99% of machines anyway. And anyone "ambitious" enough to consider it should be able to handle downloading one extra file and installing it.

**Comment by phkahler** [comment]
> @phkahler do you think I should change the extrude icon as suggested here [#857 (comment)](https://github.com/solvespace/solvespace/issues/857#issuecomment-804415270) ?

No. The way it is drawn now seems to reflect how it works better IMHO even if that's not the orientation your part ends up, it's how a sketch is extruded - out of the plane toward the user.

**Comment by phkahler** [comment]
>  The MSVC runtime distributable dependency of the OpenMP build is already installed on 99% of machines anyway.

Yes, I'm kind of regretting the 2 builds for Windows. It's not our problem if Microsoft doesn't ship all their components with every Windows install. It's common enough that we don't need to do that.

**Comment by ruevs** [comment]
The icons are updated https://github.com/solvespace/solvespace/pull/1004. 

After looking through the open issues (not all 168 ;-) it seems to me that #935, #960 and #893 should be resolved (one way or another) before 3.0. They are all on Linux.

Also the discussion on https://github.com/solvespace/solvespace/pull/997 and [in the forum](https://solvespace.com/forum.pl?action=viewthread&parent=3601&tt=1617634061) where I am involved since I last touched the code but it is also on Linux.

Maybe #963 as well but it is mostly an interesting/weird corner case.

I don't see any pull requests that would be for 3.0.

**Comment by phkahler** [comment]
#935 and #960 have not been reproduced have they? I agree that crashes are bad, but are we stuck for now or is there a way to get to the bottom of them?

**Comment by robnee** [comment]
> #935 and #960 have not been reproduced have they? I agree that crashes are bad, but are we stuck for now or is there a way to get to the bottom of them?

These are reported against Debian.  I've only tested against RHEL/Centos/Fedora and cannot reproduce on those.

**Comment by phkahler** [comment]
If there are no objections I'd like to do 3.0 final now.

**Comment by robnee** [comment]
I've updated the reference page in solvespace-web for a number of 3.0 features as well.  It failed a couple of link checks that I think are false positives but need someone to confirm.

**Comment by rpavlik** [comment]
Think that is fine by me - Debian is stuck at an earlier RC anyway due to freezes, and I'll likely be able to pull in the fix for that file-open bug when we figure it out once the freezes lift.  Thanks everybody for your hard work on wrapping up this release!

**Comment by phkahler** [comment]
@ppd  I made a tag/release for v3.0 which created a new entry on the downloads page but it didn't produce the executables.  Help?

**Comment by Symbian9** [comment]
>  I made a tag/release for v3.0 which created a new entry on the downloads page but it didn't produce the executables. Help?

Guess, You may just copy binary from latest `edge` builds and *add them manually as assets* to `v3.0` release; just release version number to each name:

- https://github.com/solvespace/solvespace/releases/download/edge/solvespace-openmp.exe
  - `solvespace-3.0-openmp.exe`
- https://github.com/solvespace/solvespace/releases/download/edge/SolveSpace.dmg
  - `SolveSpace-3.0.dmg`
- https://github.com/solvespace/solvespace/releases/download/edge/solvespace.exe
  - `solvespace-3.0.exe`

Also, add `v3.0` release description notes (key changes, link to full CHANGELOG, etc.):

**Comment by ppd** [comment]
@phkahler I think I fucked up the workflow. It skips the build jobs because it also skips the first "cancel previous build" job for release events. I'm sorry. I'll fix it, and we can copy the assets manually.

**Comment by Symbian9** [comment]
> we can copy the assets manually.

Wait, no. Is not release builds should NOT include git commit info in "About" dialog?

Release builds should include `This is SolveSpace 3.0` (without git commit infor + drop out `version` word forever), instead of `This is SolveSpace version 3.0~{git-commit}`.


I highly recommend to ***cancel/remove actual*** `v3.0` release and re-release it + add an extra commit which should force to show `This is SolveSpace 3.0` in "About" dialog for the release.

**Comment by phkahler** [comment]
@ppd thank you for your help with that!

@jwesthues can you have a quick look at the web site updates and push them live?

**Comment by jwesthues** [comment]
Thanks, and thanks all. I've pulled to live; please check and let me know if you see any trouble.

**Comment by Symbian9** [comment]
@jwesthues, may you edit release description — add some *release notes* and link to [`CHANGELOG.md`](https://github.com/solvespace/solvespace/blob/master/CHANGELOG.md)

- https://github.com/solvespace/solvespace/releases/tag/v3.0

Also, please, add links to Linux builds:

- Snap builds on Snapcraft
  - https://snapcraft.io/solvespace

- @AlexanderP's PPA (where fresh SolveSpace packages for Debian & Ubuntu just rolled out):
  - https://notesalexp.org/packages/en/buster/amd64/solvespace/
    - https://notesalexp.org/packages/en/buster/amd64/libslvs1/
    - https://notesalexp.org/packages/en/buster/amd64/libslvs1-dev/
  - https://launchpad.net/~alex-p/+archive/ubuntu/solvespace/+packages

**Comment by ruevs** [comment]
> > we can copy the assets manually.
> 
> Wait, NO! Is not release builds should NOT include git commit info in "About" dialog?
> 
No it "is not". The releases will include the hash. This was already discussed.

**Comment by Symbian9** [comment]
> he releases will include the hash. This was already discussed.

Then there should be next info in about dialog: `This is a SolveSpace 3.0~0e0b025 (release)`

**Comment by ppd** [comment]
Oh no, GitHub created a lightweight tag instead of an annotated one through its UI. 

I tested pushing an annotated tag over the lightweight tag in my fork (https://github.com/ppd/solvespace/releases/tag/v3.0), and GitHub retained the release. So that seems to be a good option. Shall we do the same here?

**Comment by Symbian9** [comment]
> So that seems to be a good option.

@ppd, Also drop out `version` word from "About" dialog.

**Comment by phkahler** [comment]
@Symbian9 we are not changing the about dialog, or anything else at this point.

**Comment by phkahler** [comment]
> Oh no, GitHub created a lightweight tag instead of an annotated one through its UI.

Whatever you think is best ;-)

**Comment by Symbian9** [comment]
> we are not changing the about dialog, or anything else at this point.

Okay, but I just recommend do that for next releases.

BTW, Edit `v3.0` *release title* & *release description* on GitHub according [my previous comment](https://github.com/solvespace/solvespace/issues/772#issuecomment-822080229)

**Comment by phkahler** [comment]
@ppd I also created a branch "version3" in case we want to back port fixes prior to next major release. Let me know if that branch should be at a different point in the history (or just move it).

**Comment by Symbian9** [comment]
> I also created a branch `version3` in case we want to back port fixes

@ppd, Good, but why not just name this branch as `3.x`? (as it already did for `2.x`)

Guess, fixes should be going into `master` for the next `v3.1` bugfix/minor release (backporting into already released `v3.0` is a wrong way because after release `v3.0` should not be re-released anymore — it just what is release is).

Think, its too early to set `master` for `v4.0`, as I expect there would be one or few minor releases of *SolveSpace* `3.x`.
